kitsune/messages/views.py

```diff
@@ -179,13 +197,37 @@ def preview_async(request):
 
 
 def _add_recipients(msg):
+    """Process and attach recipient information to a message object.
+
+    This helper function calculates recipient counts and assigns recipient-related
+    attributes to the message object for both individual users and groups.
+
+    Args:
+        msg: An OutboxMessage object to process.
+
+    Returns:
+        The modified message object with the following attributes set:
+            - recipients_count: Number of individual recipients
+            - to_groups_count: Number of group recipients
+            - recipient: The single recipient (if exactly one), else None
+            - recipient.is_bot_user: Boolean indicating if recipient is SumoBot
+            - to_groups: List of recipient groups with prefetched profiles
+
+    Note:
+        The function assumes msg.to and msg.to_group are valid related fields
+        on the message object.
+    """
     # Set the counts based on the lists
     msg.recipients_count = msg.to.all().count()
     msg.to_groups_count = msg.to_group.all().count()
 
     # Assign the recipient based on the number of recipients
     msg.recipient = msg.to.all()[0] if msg.recipients_count == 1 else None
 
+    # Set is_bot_user flag only if there's a recipient with a username
+    if msg.recipient and hasattr(msg.recipient, "username"):
```

Comment:
There is already a property in the `Profille` model that checks for system accounts, `is_system_account`